### PR TITLE
Get rid of Safe Views, use serde_skip

### LINKED ISSUE #2712: Use serde(skip) attribute for sensitive fields
We currently use separate structs like `CommunitySafe` in order to avoid that sensitive data like federation private keys are exposed in API responses. In fact this seems completely unnecessary, we can remove these "safe" structs and use [serde(skip)](https://serde.rs/field-attrs.html#skip) instead.

Similarly the trait `DeletableOrRemovable` can be replaced using [serde(skip_serializing_if)](https://serde.rs/field-attrs.html#skip_serializing_if) and a function which checks if the item was removed/deleted.

### PR DESCRIPTION
**Pull request by dessalines**
- Also change the ViewToVec, to work with non-vector cases. Might be necessary in preparation for #2763
- Fixes #2712

The federation tests are broken until the Safe versions of community, person, localuser, are removed from lemmy-ui.

### CODE REVIEW

File: crates/db_schema/src/source/community.rs [hunk 1]
```diff
@@ -19,8 +19,10 @@ pub struct Community {
   pub nsfw: bool,
   pub actor_id: DbUrl,
   pub local: bool,
+  #[serde(skip)]
   pub private_key: Option<String>,
   pub public_key: String,
+  #[serde(skip)]
```
Comment (dessalines) [inline crates/db_schema/src/source/community.rs:25]: The main changes are in this folder.

File: crates/db_schema/src/source/local_user.rs [hunk 2]
```diff
@@ -10,6 +10,7 @@ use typed_builder::TypedBuilder;
 pub struct LocalUser {
   pub id: LocalUserId,
   pub person_id: PersonId,
+  #[serde(skip)]
   pub password_encrypted: String,
```
Comment (dessalines) [inline crates/db_schema/src/source/local_user.rs:14]: Not sure if anything else should be skipped. This is only fetchable by that user anyway.

File: crates/db_schema/src/source/person.rs [hunk 3]
```diff
@@ -18,8 +18,10 @@ pub struct Person {
   pub actor_id: DbUrl,
   pub bio: Option<String>,
   pub local: bool,
+  #[serde(skip)]
   pub private_key: Option<String>,
   pub public_key: String,
+  #[serde(skip)]
```
Comment (dessalines) [inline crates/db_schema/src/source/person.rs:24]: The urls like inbox_url, etc can maybe be skipped?

Comment (Nutomic) [reply crates/db_schema/src/source/person.rs:24]: Yes and public key as well.

File: crates/db_schema/src/source/local_user.rs (hunk 2 above)
Comment (Nutomic) [reply crates/db_schema/src/source/local_user.rs:14]: Looks good.

File: crates/db_schema/src/source/community.rs (hunk 1 above)
Comment (Nutomic) [reply crates/db_schema/src/source/community.rs:25]: Also skip public key and inboxes.

**Comment by Nutomic** [comment]
In the Activitypub code there are a few places where it reads Person or Community from the database, because *Safe variants have the wrong type and are missing data. If you want to go through it, you can get rid of them, but its not important.

File: crates/db_schema/src/source/person.rs (hunk 3 above)
Comment (dessalines) [reply crates/db_schema/src/source/person.rs:24]: Doesn't public key get serialized to be sent to other instances?

Comment (Nutomic) [reply crates/db_schema/src/source/person.rs:24]: Yes but with Activitypub structs in crates/apub/src/protocol, not with these api structs here. So no need to include it.

**Comment by dessalines** [comment]
Another side note: I'd *assumed* this was a breaking change to rename all the `CommunitySafe` to `Community` and such for the API, but it turns out its not, as evidenced by the old lemmy-js-client version passing every test. Clients don't care what the interfaces are called, only the fields within the interfaces, and this didn't change any of those.

**Comment by Nutomic** [comment]
Right but removing fields like public key from api responses is breaking, even though no one really used them. Anyway we just need to avoid cherry picking this onto release branch.

**Review by Nutomic — APPROVED** [review]

**Pull request merged**